src/Transient_Command.php

```diff
@@ -221,53 +223,166 @@ public function type() {
 	 */
 	private function delete_expired() {
 		global $wpdb;
 
-		// Always delete all transients from DB too.
-		$time = current_time('timestamp');
 		$count = $wpdb->query(
-			"DELETE a, b FROM $wpdb->options a, $wpdb->options b WHERE
-			a.option_name LIKE '\_transient\_%' AND
-			a.option_name NOT LIKE '\_transient\_timeout\_%' AND
-			b.option_name = CONCAT( '_transient_timeout_', SUBSTRING( a.option_name, 12 ) )
-			AND b.option_value < $time"
+			$wpdb->prepare(
+				"DELETE a, b FROM {$wpdb->options} a, {$wpdb->options} b
+					WHERE a.option_name LIKE %s
+					AND a.option_name NOT LIKE %s
+					AND b.option_name = CONCAT( '_transient_timeout_', SUBSTRING( a.option_name, 12 ) )
+					AND b.option_value < %d",
+				Utils\esc_like( '_transient_' ) . '%',
+				Utils\esc_like( '_transient_timeout_' ) . '%',
+				time()
+			)
 		);
 
+		if ( ! is_multisite() ) {
+			// Non-Multisite stores site transients in the options table.
+			$count += $wpdb->query(
+				$wpdb->prepare(
+					"DELETE a, b FROM {$wpdb->options} a, {$wpdb->options} b
+						WHERE a.option_name LIKE %s
+						AND a.option_name NOT LIKE %s
+						AND b.option_name = CONCAT( '_site_transient_timeout_', SUBSTRING( a.option_name, 17 ) )
+						AND b.option_value < %d",
+					Utils\esc_like( '_site_transient_' ) . '%',
+					Utils\esc_like( '_site_transient_timeout_' ) . '%',
+					time()
+				)
+			);
+		} else {
+			// Multisite stores site transients in the sitemeta table.
+			$count += $wpdb->query(
+				$wpdb->prepare(
+					"DELETE a, b FROM {$wpdb->sitemeta} a, {$wpdb->sitemeta} b
+						WHERE a.meta_key LIKE %s
+						AND a.meta_key NOT LIKE %s
+						AND b.meta_key = CONCAT( '_site_transient_timeout_', SUBSTRING( a.meta_key, 17 ) )
+						AND b.meta_value < %d",
+					Utils\esc_like( '_site_transient_' ) . '%',
+					Utils\esc_like( '_site_transient_timeout_' ) . '%',
+					time()
+				)
+			);
+		}
+
+		// The above queries delete the transient and the transient timeout
+		// thus each transient is counted twice.
+		$count = $count / 2;
+
 		if ( $count > 0 ) {
-			WP_CLI::success( "$count expired transients deleted from the database." );
+			WP_CLI::success(
+				sprintf(
+					'%d expired %s deleted from the database.',
+					$count,
+					Utils\pluralize( 'transient', $count )
+				)
+			);
 		} else {
-			WP_CLI::success( "No expired transients found." );
+			WP_CLI::success( 'No expired transients found.' );
 		}
 
 		if ( wp_using_ext_object_cache() ) {
-			WP_CLI::warning( 'Transients are stored in an external object cache, and this command only deletes those stored in the database. You must flush the cache to delete all transients.');
+			WP_CLI::warning( 'Transients are stored in an external object cache, and this command only deletes those stored in the database. You must flush the cache to delete all transients.' );
 		}
 	}
 
 	/**
 	 * Deletes all transients.
+	 *
+	 * Only deletes the transients from the database.
```

Comment:
This should behave the same as other commands and should delete site transients only when the  `--network` flag is set.